crates/oci-distribution/src/reference.rs

```diff
@@ -90,10 +89,7 @@ impl Reference {
 
     /// full_name returns the full repository name and path.
     fn full_name(&self) -> String {
-        let mut path = PathBuf::new();
-        path.push(self.registry());
-        path.push(self.repository());
-        path.to_str().unwrap_or("").to_owned()
+        format!("{}/{}", self.registry(), self.repository())
```

Comment:
It is possible that `self.registry()` may return an empty string, which will result in a string like `/foo`. The previous code caught this scenario because an empty string would return `foo`.

One way to fix might be to check if `self.registry()` returns an empty string, and only join the two together if it exists.

It might be worth writing a unit test just to make sure I'm not totally wrong that the old code did indeed work as I described...

Reply:
Good catch - thanks!

Reply:
Fixed and test added - please take another look when convenient